### DISCUSSION TITLE
"No healthy upstream" error

### DISCUSSION THREAD

**Discussion by HideoKuze** [Budibase/budibase#5081] [Help]
Recently my AWS EC2 instanece changed it's IP address. On doing so I needed to initialize another Budibase project. In doing so it encountered an error creating images like "bbapps" and "bbworker" because those names already existed from my old Budibase installation. So I just renamed them, then installation proceeded successfully. I later decided to just delete the old images, assuming I didn't need them. Now I every time I visit my website I get the error "No health upstream". Do I need to do another install? Will this effect my apps?

Docker ps:

![image](https://user-images.githubusercontent.com/5965711/159969261-bd402ea0-014d-4aaa-9f31-bb232fd42631.png)

**Comment by shogunpurple** [comment]
Hi @HideoKuze,

This shouldn't affect your apps, given that your docker volumes are being used which contain all your data. You didn't delete the volumes, just the images.

Can I ask why you needed to initialise a new budibase project? Were you moving to a new EC2 instance?

Also how are you accessing your website? Just through the IP address of the EC2 instance in the browser? I'm assuming so just seeing that port 80 is exposed.

**Reply by HideoKuze** [reply]
It wasn't a new EC2 instance but a new IP address, so I just initialized a new project and it was working fine. Until I removed the old images. 

I am accessing it through my domain which has the IP address as it's A record, which was also working fine. Should I remove all the images and reinitialise?

**Reply by shogunpurple** [reply]
If the IP changed, it seems like the second step is the only thing you needed to do there - to point your domain to the new IP with your A record.

When you run a `docker-compose logs` do you see any error messages?

**Reply by HideoKuze** [reply]
Sorry if I was unclear, I changed the IP, then I reinitialised, and it was working fine. It's not until I deleted the old images (that I had renamed to facilitate the new initialization) that this error occurred. 

This is the last bit of messages I saw that displayed an error:
![image](https://user-images.githubusercontent.com/5965711/159974169-cda4a987-78a4-4082-8722-bdd2eca2cf58.png)

**Reply by HideoKuze** [reply]
It's almost as if that somewhere one or more of the images is looking for the old bbapps/worker? I'm okay will deleting all the images and reinstalling as long as that effects nothing else

**Reply by shogunpurple** [reply]
@HideoKuze - it seems like couch is running on `any` as its hostname, did you update your `docker-compose.yaml` file with more configuration to cause this?

**Reply by HideoKuze** [reply]
Made no changes, when I reinitialized a new project though I didn't delete any of the old directories. Maybe that could be an issue?

**Reply by shogunpurple** [reply]
@HideoKuze would you be able to paste the contents of your `docker-compose.yaml` in here please?

**Reply by HideoKuze** [reply]
`version: "4"

# optional ports are specified throughout for more advanced use cases.

services:
  app-service:
    restart: always
    image: budibase.docker.scarf.sh/budibase/apps
    container_name: bbapps
    environment:
      SELF_HOSTED: 1
      COUCH_DB_URL: http://${COUCH_DB_USER}:${COUCH_DB_PASSWORD}@couchdb-service:5984
      WORKER_URL: http://worker-service:4003
      MINIO_URL: http://minio-service:9000
      MINIO_ACCESS_KEY: ${MINIO_ACCESS_KEY}
      MINIO_SECRET_KEY: ${MINIO_SECRET_KEY}
      INTERNAL_API_KEY: ${INTERNAL_API_KEY}
      BUDIBASE_ENVIRONMENT: ${BUDIBASE_ENVIRONMENT}
      PORT: 4002
      JWT_SECRET: ${JWT_SECRET}
      LOG_LEVEL: info
      SENTRY_DSN: https://a34ae347621946bf8acded18e5b7d4b8@o420233.ingest.sentry.io/5338131
      ENABLE_ANALYTICS: "true"
      REDIS_URL: redis-service:6379
      REDIS_PASSWORD: ${REDIS_PASSWORD}
    depends_on:
      - worker-service
      - redis-service

  worker-service:
    restart: always
    image: budibase.docker.scarf.sh/budibase/worker
    container_name: bbworker
    environment:
      SELF_HOSTED: 1
      PORT: 4003
      CLUSTER_PORT: ${MAIN_PORT}
      JWT_SECRET: ${JWT_SECRET}
      MINIO_ACCESS_KEY: ${MINIO_ACCESS_KEY}
      MINIO_SECRET_KEY: ${MINIO_SECRET_KEY}
      MINIO_URL: http://minio-service:9000
      APPS_URL: http://app-service:4002
      COUCH_DB_USERNAME: ${COUCH_DB_USER}
      COUCH_DB_PASSWORD: ${COUCH_DB_PASSWORD}
      COUCH_DB_URL: http://${COUCH_DB_USER}:${COUCH_DB_PASSWORD}@couchdb-service:5984
      SENTRY_DSN: https://a34ae347621946bf8acded18e5b7d4b8@o420233.ingest.sentry.io/5338131
      INTERNAL_API_KEY: ${INTERNAL_API_KEY}
      REDIS_URL: redis-service:6379
      REDIS_PASSWORD: ${REDIS_PASSWORD}
    depends_on:
      - redis-service
      - minio-service
      - couch-init

  minio-service:
    restart: always
    image: minio/minio
    volumes:
      - minio_data:/data
    environment:
      MINIO_ACCESS_KEY: ${MINIO_ACCESS_KEY}
      MINIO_SECRET_KEY: ${MINIO_SECRET_KEY}
      MINIO_BROWSER: "off"
    command: server /data
    healthcheck:
      test: ["CMD", "curl", "-f", "http://localhost:9000/minio/health/live"]
      interval: 30s
      timeout: 20s
      retries: 3

  proxy-service:
                                                                                                                                                                                                                                                             1,11          Top

 couchdb-service:
    restart: always
    image: ibmcom/couchdb3
    environment:
      - COUCHDB_PASSWORD=${COUCH_DB_PASSWORD}
      - COUCHDB_USER=${COUCH_DB_USER}
    volumes:
      - couchdb3_data:/opt/couchdb/data

  couch-init:
    image: curlimages/curl
    environment:
      PUT_CALL: "curl -u ${COUCH_DB_USER}:${COUCH_DB_PASSWORD} -X PUT couchdb-service:5984"
    depends_on:
      - couchdb-service
    command: ["sh","-c","sleep 10 && $${PUT_CALL}/_users && $${PUT_CALL}/_replicator; fg;"]

  redis-service:
    restart: always
    image: redis
    command: redis-server --requirepass ${REDIS_PASSWORD}
    volumes:
      - redis_data:/data

  watchtower-service:
    image: containrrr/watchtower
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock
    command: --debug --http-api-update bbapps bbworker bbproxy
    environment:
      - WATCHTOWER_HTTP_API=true
      - WATCHTOWER_HTTP_API_TOKEN=budibase
      - WATCHTOWER_CLEANUP=true
    labels:
      - "com.centurylinklabs.watchtower.enable=false"


volumes:
  couchdb3_data:
    driver: local
  minio_data:
    driver: local
  redis_data:
    driver: local

`

**Reply by shogunpurple** [reply]
@HideoKuze this file doesn't seem to be valid.

```
proxy-service:
1,11 Top
```

It appears you haven't pasted it properly. Could you format it, or put it into a service like pastebin? Thanks

**Reply by HideoKuze** [reply]
I remember I also had to go into the images `config.v2.json` and `hostingconfig.json` files to edit their port binding manually. Since when I initialised the project they were not mapped to any ports. I'm wondering if maybe there's something else in that file missing? Like I said if I can delete all the files created at initialization and redo it all without effecting the apps/volumes im okay with it

**Reply by shogunpurple** [reply]
@HideoKuze - I don't know what those files are. Where did you find them? It seems like you are using some additional, non-budibase or EC2 related software.

If you want to start again from scratch, without removing your volumes - you should be able to do:

`docker-compose down`

This will not remove volumes by default.

**Reply by HideoKuze** [reply]
@shogunpurple Sorry here you go: https://pastebin.com/KN7w7S0r

**Reply by shogunpurple** [reply]
I see it - you are using an old version of the `docker-compose.yaml` that's still running the envoy proxy.

Pull this one: https://raw.githubusercontent.com/Budibase/budibase/master/hosting/docker-compose.yaml

and run `docker-compose up -d` again.

**Reply by HideoKuze** [reply]
Awesome thanks for finding that. It's telling me that bbworker is already in use, im guessing I can just rename it then

**Reply by HideoKuze** [reply]
I renamed them and it's working fine now, thank you
